axelrod/moran.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+from collections import Counter
+import random
+
+import numpy as np
+
+from .deterministic_cache import DeterministicCache
+from .match import Match
+from .player import Player
+from .random_ import randrange
+
+
+def fitness_proportionate_selection(scores):
+    """Randomly selects an individual proportionally to score."""
+    csums = np.cumsum(scores)
+    total = csums[-1]
+    r = random.random() * total
+
+    for i, x in enumerate(csums):
+        if x >= r:
+            return i
```

Comment:
Hmm.. I'm not sure which is more simple. `scores` will still have to be normalized, something like:
```
np.random.choice(len(scores), 1, p=np.array(scores) / sum(scores)
```
Technically scores could be all zeros in my implementation -- that shouldn't happen for IPD but it could for another game matrix. So we'd need a check or try/except block for that, and it might be better as is.